### ISSUE TITLE
Cannot override writable attribute with a final one

### ISSUE THREAD

**Issue by fr0der1c** [ambv/flake8-mypy#28]
The below code should pass the check:
```python
class MongoDAOBase(abc.ABC):
    collection_name: str = NotImplemented

    @classmethod
    @abc.abstractmethod
    def create_index(cls) -> None:
        pass


class PrivacySettingsDAO(MongoDAOBase):
    collection_name: Final = "privacy_settings"

    @classmethod
    def create_index(cls) -> None:
        pass
```

However, a `T484 Cannot override writable attribute "collection_name" with a final one`is raised. This problem seems to be introduced recently. My local version (which is older) of mypy does not complain, but it fails in CI environment (which gets later version).
